src/extensions/layout-selector/test/layout-selector.cypress.js

```diff
@@ -87,4 +87,25 @@ describe( 'Extension: Layout Selector', () => {
 		cy.get( '.coblocks-layout-selector-modal' ).should( 'exist' );
 	} );
 
+	it( 'does not open modal when editing a draft post', () => {
+		helpers.goTo( '/wp-admin/post-new.php?post_type=page' );
+		helpers.disableGutenbergFeatures();
+
+		cy.get( '.coblocks-layout-selector__layout' ).first().click();
+		cy.get( '.editor-post-save-draft' ).click();
+
+		// Reload the post.
+		let postID;
+		cy.window()
+			.then( (win) => {
+				postID = win.wp.data.select( 'core/editor' ).getCurrentPostId()
+			} )
+			.then( () => {
+				helpers.goTo( `/wp-admin/post.php?post=${postID}&action=edit` );
+
+				helpers.disableGutenbergFeatures();
+				cy.get( '.coblocks-layout-selector-modal' ).should( 'not.exist' );
```

Comment:
Some of this logic can be simplified by removing the single-use variable.
```javascript
	let postID;
	cy.window()
			.then( (win) => {
				postID = win.wp.data.select( 'core/editor' ).getCurrentPostId()
			} )
			.then( () => {
				helpers.goTo( `/wp-admin/post.php?post=${postID}&action=edit` );

				helpers.disableGutenbergFeatures();
				cy.get( '.coblocks-layout-selector-modal' ).should( 'not.exist' );
			} );
```

Could Be

```javascript
	cy.window()
			.then( (win) => {
				helpers.goTo( `/wp-admin/post.php?post=${win.wp.data.select( 'core/editor' ).getCurrentPostId()}&action=edit` );

				helpers.disableGutenbergFeatures();
				cy.get( '.coblocks-layout-selector-modal' ).should( 'not.exist' );
			} );
```

Reply:
Although this may look unecessary, it's crucial that we wait for the `getCurrentPostId()` to resolve (it's a Promise) before attempting to use it. Combining the `then()` callbacks does not give us the post id in time.